plugin/storage/blackhole/factory.go

```diff
@@ -40,7 +41,7 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 
 // CreateSpanReader implements storage.Factory
 func (f *Factory) CreateSpanReader() (spanstore.Reader, error) {
-	return f.store, nil
+	return storageMetrics.NewReadMetricsDecorator(f.store, f.metricsFactory), nil
```

Comment:
@yurishkuro all the storage implementations already hold a `metrics.Factory` so can we just use those to decorate the readers like I did here?

Reply:
Yes, but we also need to make sure all storage implementations use a consistent namespace

Reply:
@yurishkuro they're being passed in from [here](https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/extension/jaegerstorage/extension.go#L120) for v2 with the same namespace - is that what you meant?

Reply:
We need to understand how this might affect the existing behavior and backwards compatibility. Can you check how the metrics are reported today in v1?

Reply:
@yurishkuro From all-in-one, the metric factory for the query service is namespaced under `jaeger/query` (ref: https://github.com/jaegertracing/jaeger/blob/main/cmd/all-in-one/main.go#L91) and this is used in the decorator for the span reader (https://github.com/jaegertracing/jaeger/blob/main/cmd/all-in-one/main.go#L227). Same is done in the v2 query extension (ref: https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/extension/jaegerquery/server.go#L56-L57).  It doesn't look like the archive factory is wrapped at all (ref: https://github.com/jaegertracing/jaeger/blob/main/cmd/query/app/querysvc/query_service.go#L136). However, we have Cassandra that has a dedicated metrics factory for the archive reader (https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/cassandra/factory.go#L137).

Reply:
Tip: it's easiest to see the actual metrics emitted (from metrics endpoint) where both the name and labels are already resolved

Reply:
The metrics factory that is passed into the storage factories is namespaced under just `jaeger` (https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/extension/jaegerstorage/extension.go#L120). When creating the spanReader inside the factories, we could namespace the metrics factory under `query` and we should be able to achieve backwards compatible behaviour. What do you think?

Reply:
@yurishkuro How can I see the actual metrics? Would it be at the prometheus endpoint?

Reply:
Yes, Prometheus endpoint. You need to start binaries with different storage types to see all metrics.

Reply:
@yurishkuro what should I query on prometheus to see the metrics?

Reply:
@yurishkuro Got it - I tried adding the following config to `docker-compose/monitor/jaeger-v2-config.yml`
```
  telemetry:
    metrics:
      level: detailed
      address: 0.0.0.0:8888
```
When I try running the example, I see `jaeger-1      | 2024-11-23T22:47:11.986Z        info    telemetry/metrics.go:70 Serving metrics {"address": "0.0.0.0:8888", "metrics level": "Detailed"}` printed in the logs but I get an error when trying to access that endpoint. Am I missing something? 
```
curl http://localhost:8888/metrics
curl: (7) Failed to connect to localhost port 8888 after 5 ms: Connection refused
```

Reply:
@yurishkuro This is what I'm seeing
```
# HELP jaeger_query_latency 
# TYPE jaeger_query_latency histogram
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="0"} 0
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="5"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="10"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="25"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="50"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="75"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="100"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="250"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="500"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="750"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="1000"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="2500"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="5000"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="7500"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="10000"} 2
jaeger_query_latency_bucket{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z",le="+Inf"} 2
jaeger_query_latency_sum{operation="find_traces",result="ok",service_instance_id="61b297e0-bf9b-415d-9926-e22ce48cf4e5",service_name="jaeger",service_version="git-commit=7b4631c4d54f12f4194e701f50b66af1f1e0d499, git-version=v2.0.0, build-date=2024-11-23T01:19:44Z"} 0.023833042
```

Reply:
This is for the memory storage - so it looks like there's no storage specific namespace here

Reply:
see https://github.com/jaegertracing/jaeger/pull/6243/files#r1855475987